ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py

```diff
@@ -972,3 +972,60 @@ def test_numpy_triangular(
         right=right,
         size=size,
     )
+
+
+# noncentral_f - numpy
+@handle_frontend_test(
+    fn_tree="numpy.random.noncentral_f",
+    input_dtypes=helpers.get_dtypes("float"),
+    dfnum=st.floats(
+        allow_nan=False,
+        allow_infinity=False,
+        width=32,
+        min_value=1,
+        max_value=1000,
+        exclude_min=True,
+    ),
+    dfden=st.floats(
+        allow_nan=False,
+        allow_infinity=False,
+        width=32,
+        min_value=1,
+        max_value=1000,
+        exclude_min=True,
+    ),
+    nonc=st.floats(
+        allow_nan=False,
+        allow_infinity=False,
+        width=32,
+        min_value=0,
+        max_value=1000,
+    ),
+    size=helpers.get_shape(allow_none=False),
```

Comment:
Could you please add this data-generation part in a helper function? This will make the code more cleaner :sparkles:

Reply:
@hello-fri-end ,

I have moved the data generation to a function called generate_random_data()